modules/rostests/apitests/shell32/menu.cpp

```diff
@@ -236,54 +238,55 @@ class CMenuCallback : public CUnknownBase<IShellMenuCallback>
         m_testsCount = testsCount;
     }
 
-   void SetTest(int i)
-   {
-       m_iTest = i;
-   }
+    void SetTest(int i)
+    {
+        m_iTest = i;
+    }
 
-   HRESULT STDMETHODCALLTYPE CallbackSM(LPSMDATA psmd, UINT uMsg, WPARAM wParam, LPARAM lParam)
-   {
+    HRESULT STDMETHODCALLTYPE CallbackSM(LPSMDATA psmd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+    {
        /*trace ("callback type %d\n", uMsg);*/
 
        /*
         * it seems callback 0x10000000 is called for every item added so
-        * we will ignore consecutive callbacks of this type
+        * we will ignore callbacks of this type
         * Note: this callback is invoked by shell32.dll!CMenuSFToolbar::_FilterPidl
         */
-       if (uMsg == 0x10000000 && m_results[m_iCallback-1].uMsg == 0x10000000)
-       {
-           return S_OK;
-       }
-
-       m_iCallback++;
-       if (m_iCallback > m_testsCount)
-       {
-           ok(0, "Got more callbacks than expected! (%d not %d). uMsg: %d\n", m_iCallback, m_testsCount, uMsg);
-           return S_OK;
-       }
-
-       struct _test_info *result = &m_results[m_iCallback-1];
-
-       ok(psmd != NULL, "Got NULL psmd\n");
-       ok(m_iTest == result->iTest, "Wrong test number (%d not %d)\n", m_iTest, result->iTest);
-       ok(result->uMsg == uMsg, "%d: Got wrong uMsg (%d instead of %d)\n", m_iCallback, uMsg, result->uMsg);
-
-       if(uMsg == SMC_CREATE)
-       {
-           ok(psmd->dwFlags == 0, "wrong dwFlags\n");
-           ok(psmd->dwMask == 0, "wrong dwMask\n");
-           ok(psmd->hmenu == 0, "wrong hmenu\n");
-           ok(psmd->hwnd == 0, "wrong hwnd\n");
-           ok(psmd->punk != NULL, "punk is null\n");
-       }
-
-       if (uMsg == SMC_GETSFOBJECT)
-       {
-           ok(psmd->psf != 0, "wrong dwFlags\n");
-       }
-
-       return S_FALSE;
-   }
+        if (uMsg == 0x10000000 && m_results[m_iCallback-1].uMsg == 0x13)
+        {
+            return S_OK;
+        }
+
+        m_iCallback++;
+        if (m_iCallback > m_testsCount)
+        {
+            ok(0, "Got more callbacks than expected! (%d not %d). uMsg: %d\n", m_iCallback, m_testsCount, uMsg);
```

Comment:
```suggestion
            ok(FALSE, "Got more callbacks than expected! (%d not %d). uMsg: %d\n", m_iCallback, m_testsCount, uMsg);
```
Looks like a candidate for a `skip(...)` instead.

Reply:
Not a good candidate for skip, the test is explicitly counting the number of callbacks here.